public/app/features/provisioning/RepositoryStatusPage.tsx

```diff
@@ -1,14 +1,131 @@
+import { useMemo, useState } from 'react';
+import { useParams } from 'react-router-dom-v5-compat';
+
+import {
+  CellProps,
+  Column,
+  EmptyState,
+  FilterInput,
+  InteractiveTable,
+  LinkButton,
+  Spinner,
+  Stack,
+  Text,
+  TextLink,
+} from '@grafana/ui';
 import { Page } from 'app/core/components/Page/Page';
 
+import { useGetRepositoryStatusQuery, useListRepositoryFilesQuery } from './api';
+import { PROVISIONING_URL } from './constants';
+
 export default function RepositoryStatusPage() {
+  const { name = '' } = useParams();
+  const query = useGetRepositoryStatusQuery({ name });
+
+  //@ts-expect-error TODO add error types
+  const notFound = query.isError && query.error?.status === 404;
   return (
     <Page
       navId="provisioning"
-      pageNav={{ text: 'Repository status', subTitle: 'Check the status of a configured repository.' }}
+      pageNav={{
+        text: query.data?.spec.title ?? 'Repository Status',
+        subTitle: 'Check the status of configured repository.',
+      }}
     >
-      <Page.Contents>
-        <div>Repository status</div>
+      <Page.Contents isLoading={false}>
+        {notFound ? (
+          <EmptyState message={`Repository not found`} variant="not-found">
+            <Text element={'p'}>Make sure the repository config exists in the configuration file.</Text>
+            <TextLink href={PROVISIONING_URL}>Back to repositories</TextLink>
+          </EmptyState>
+        ) : (
+          <FilesTable name={name} />
+        )}
       </Page.Contents>
     </Page>
   );
 }
+
+type FileDetails = {
+  path: string;
+  size: string;
+  hash: string;
+};
+
+interface FilesTableProps {
+  name: string;
+}
+
+type Cell<T extends keyof FileDetails = keyof FileDetails> = CellProps<FileDetails, FileDetails[T]>;
+
+function FilesTable({ name }: FilesTableProps) {
+  const query = useListRepositoryFilesQuery({ name });
+  const [searchQuery, setSearchQuery] = useState('');
+  const data = [...(query.data?.files ?? [])].filter((file) =>
+    file.path.toLowerCase().includes(searchQuery.toLowerCase())
+  );
+  const columns: Array<Column<FileDetails>> = useMemo(
+    () => [
+      {
+        id: 'path',
+        header: 'Path',
+        sortType: 'string',
+      },
+      {
+        id: 'size',
+        header: 'Size',
+        sortType: 'number',
```

Comment:
Maybe we should say something about it being bytes, or convert it to a human readable file size?

Reply:
Yeah, good point!